Tests/FluentAssertions.Shared.Specs/ObjectAssertionSpecs.cs

```diff
@@ -454,6 +454,68 @@ public void When_object_is_assignable_to_the_expected_type_it_should_cast_the_re
             act.ShouldThrow<AssertFailedException>().WithMessage("*Expected*Other*Actual*");
         }
 
+        [TestMethod]
+        public void When_asserting_an_object_is_assignable_its_own_type_using_type_instance_it_should_succeed()
```

Comment:
It's my pleasure. Thanks for sharing the article, Maybe I am too focused on sticking with the existing conventions and style of the code base.

For the two comments below, I totally agree with you and I didn't like the `Invoking()` usage in the first place. However, the tests I added are already placed within a region (BeAssignableTo). For instance I can rename the first test to  `When_its_own_type_it_succeeds` removing the part you specified. Maybe `When_subjects_own_type_it_succeeds`